### DISCUSSION TITLE
Are there any plans to implement layouts?

### DISCUSSION THREAD

**Discussion by AlexNabokikh** [pop-os/shell#1630] [Q&A]
Hi!
The extension is already handy, but it would be even better if we could have the option to select layouts that are similar to those found in more traditional tiling window managers. For example, on my macOS machine, I currently use Amethyst, which offers over 15+ layouts that enable me to quickly rearrange windows using key bindings. However, I believe that having just a few layouts like "two pane," "tall," "float," and "binary space partitioning" would satisfy the needs of 99% of users. It would be great if we could incorporate these layout options into the extension. Thank you!

**Comment by arkonbob** [comment]
Yes, layouts would be great (I'm also an amethyst user, at work). For myself however,  the most important addition would be the single window or 'monacle' layout, because at present there is no way to do this in Pop Shell,  even manually (maximising a window is reverted as soon as another window is selected/focused).

**Reply by AlexNabokikh** [reply]
Yes, I just remembered that I should mention this. I often use a monocle/fullscreen layout, especially on smaller screens.

**Comment by jacobgkau** [comment]
There have been similar feature requests opened as issues in the past, such as https://github.com/pop-os/shell/issues/1057 and https://github.com/pop-os/shell/issues/1353.

The Pop!_Shell extension for GNOME Shell is being superseded by https://github.com/pop-os/cosmic-comp for tiling (and https://github.com/pop-os/cosmic-launcher for the launcher frontend) in the new [COSMIC Epoch](https://github.com/pop-os/cosmic-epoch) desktop environment, which is currently under heavy development. It's unlikely a large new feature like this would be implemented in the GNOME Shell extension at this point.

Tagging @pop-os/ux so they see the suggestion. Not sure if we want to transfer this discussion to one of the COSMIC Epoch repositories.

**Reply by AlexNabokikh** [reply]
@jacobgkau thank you for response. I checked these issues and they are not related to this request.
Layouts are part of the mainstream automatic tailing managers (awesome, i3, bspwm or yabai and amethyst on macos). Pop os has only one built-in layout, that splits every next window by half. As I said - it is already handy, but just not enough to feel it as a complete replacement for tiling managers that I mentioned. gTile or any other manual tiling managers are different type of products solving different problem.
